runner/debug/consul.go

```diff
@@ -0,0 +1,101 @@
+package debug
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/hcdiag/op"
+	"github.com/hashicorp/hcdiag/redact"
+	"github.com/hashicorp/hcdiag/runner"
+)
+
+var _ runner.Runner = ConsulDebug{}
+
+// ConsulDebugConfig is a config struct for ConsulDebug runners
+type ConsulDebugConfig struct {
+	Archive    string
+	Duration   string
+	Interval   string
+	Captures   []string
+	Redactions []*redact.Redact
+}
+
+// ConsulDebug represents a ConsulDebug runner
+type ConsulDebug struct {
+	Archive    string           `json:"archive"`
+	Duration   string           `json:"duration"`
+	Interval   string           `json:"interval"`
+	Captures   []string         `json:"captures"`
+	Redactions []*redact.Redact `json:"redactions"`
+
+	output string
+}
+
+func (d ConsulDebug) ID() string {
+	return "ConsulDebug"
+}
+
+func NewConsulDebug(cfg ConsulDebugConfig, tmpDir string, debugDuration time.Duration, debugInterval time.Duration) *ConsulDebug {
+	// Create a pseudorandom string of characters to allow >1 ConsulDebug runner without filename collisions
+	randStr := randAlphanumString(4)
+
+	dbg := ConsulDebug{
+		// No compression because the hcdiag bundle will get compressed anyway
+		Archive: "true",
+		// Use debug duration and interval
+		Duration: debugDuration.String(),
+		Interval: debugInterval.String(),
+		// Creates a subdirectory inside output dir
+		output:     debugOutputPath(tmpDir, "ConsulDebug", randStr),
+		Captures:   cfg.Captures,
+		Redactions: cfg.Redactions,
+	}
+
+	if len(cfg.Archive) > 0 {
```

Comment:
Nit: Since we're not looking for a specific character in a string, but rather, is the string set to something other than `""`, I think this would be more readable as:

```suggestion
	if cfg.Archive != "" {
```

This may just be me, but when I saw the length check, I did a double-take to try and figure out what was happening.